coresimd/arm/cmsis.rs

```diff
@@ -0,0 +1,264 @@
+#![allow(non_snake_case)]
+
+/// Extracted from [CMSIS 5]'s CMSIS/Core/Include/cmsis_gcc.h
+///
+/// [CMSIS 5]: https://github.com/ARM-software/CMSIS_5
+
+/* Core function access */
+
+/// Enable IRQ Interrupts
+///
+/// Enables IRQ interrupts by clearing the I-bit in the CPSR. Can only be
+/// executed in Privileged modes.
+#[inline(always)]
+pub unsafe fn __enable_irq() {
```

Comment:
These should use the `assert_instr` macro.

Reply:
@alexcrichton these are all in a module guarded by `cfg(target_feature = "mclass")`, so they don't technically need `#[target_feature(enable = "mclass")]`, but I think we still do always use everywhere in the library even in those classes. 

I think I would prefer to have it here as well just in case we end up moving functions around internally in the future. 

@japaric is it possible to detect cmsis at run-time?

Reply:
Yeah I think it's probably prudent to go ahead and add the attribute to manually enable it just in case.